diff --git a/Source/multi.cpp b/Source/multi.cpp
index f8200289..f758ba1a 100644
--- a/Source/multi.cpp
+++ b/Source/multi.cpp
@@ -756,11 +756,11 @@ int InitLevelType(int l)
 {
 	if (l == 0)
 		return 0;
-	if (l >= 1 && l <= 4)
+	if (l >= 1 && l <= 8)
 		return 1;
-	if (l >= 5 && l <= 8)
+	if (l >= 9 && l <= 16)
 		return 2;
-	if (l >= 9 && l <= 12)
+	if (l >= 16 && l <= 24)
 		return 3;
 
 	return 4;
diff --git a/defs.h b/defs.h
index 059de08e..0e76bd61 100644
--- a/defs.h
+++ b/defs.h
@@ -25,20 +25,20 @@
 #define MAX_SEND_STR_LEN		80
 #define MAX_SPELLS				37
 
-#define MAXDEAD					31
-#define MAXDUNX					112
-#define MAXDUNY					112
-#define MAXITEMS				127
+#define MAXDEAD					61
+#define MAXDUNX					224
+#define MAXDUNY					224
+#define MAXITEMS				254
 #define MAXBELTITEMS			8
 #define MAXLIGHTS				32
 #define MAXMISSILES				125
-#define MAXMONSTERS				200
+#define MAXMONSTERS				400
 #define MAXMULTIQUESTS			4
-#define MAXOBJECTS				127
+#define MAXOBJECTS				254
 #define MAXPORTAL				4
 #define MAXQUESTS				16
 #define MAXTHEMES				50
-#define MAXTILES				2048
+#define MAXTILES				4096
 #define MAXTRIGGERS				5
 #define MAXVISION				32
 #define MDMAXX					40
@@ -59,7 +59,7 @@
 #define NUM_TOWNERS				16
 
 // todo: enums
-#define NUMLEVELS				17
+#define NUMLEVELS				34
 #define SMITH_ITEMS				20
 #define SMITH_PREMIUM_ITEMS		6
 #define SMITH_MAX_VALUE			140000
