diff --git a/Source/multi.cpp b/Source/multi.cpp
index f8200289..7567f97a 100644
--- a/Source/multi.cpp
+++ b/Source/multi.cpp
@@ -756,11 +756,11 @@ int InitLevelType(int l)
 {
 	if (l == 0)
 		return 0;
-	if (l >= 1 && l <= 4)
+	if (l >= 1 && l <= 8)
 		return 1;
-	if (l >= 5 && l <= 8)
+	if (l >= 9 && l <= 16)
 		return 2;
-	if (l >= 9 && l <= 12)
+	if (l >= 17 && l <= 24)
 		return 3;
 
 	return 4;
diff --git a/Source/player.cpp b/Source/player.cpp
index 93989dcf..87303b9b 100644
--- a/Source/player.cpp
+++ b/Source/player.cpp
@@ -78,9 +78,9 @@ char *ClassStrTblOld[] = {
 };
 /** Maps from player_class to maximum stats. */
 int MaxStats[3][4] = {
-	{ 250, 50, 60, 100 },
-	{ 55, 70, 250, 80 },
-	{ 45, 250, 85, 80 }
+	{ 500, 100, 120, 200 },
+	{ 110, 140, 500, 160 },
+	{ 90, 500, 170, 80 }
 };
 /** Specifies the experience point limit of each level. */
 int ExpLvlsTbl[MAXCHARLEVEL] = {
diff --git a/SourceX/DiabloUI/selgame.cpp b/SourceX/DiabloUI/selgame.cpp
index e786503d..895efff6 100644
--- a/SourceX/DiabloUI/selgame.cpp
+++ b/SourceX/DiabloUI/selgame.cpp
@@ -70,6 +70,7 @@ void selgame_GameSelection_Init()
 	}
 
 	getIniValue("Phone Book", "Entry1", selgame_Ip, 128);
+	getIniValue("Phone Book", "Password1", selgame_Password, 15);
 
 	selgame_FreeVectors();
 
@@ -350,7 +351,7 @@ void selgame_Speed_Select(int value)
 
 void selgame_Password_Init(int value)
 {
-	memset(&selgame_Password, 0, sizeof(selgame_Password));
+	// memset(&selgame_Password, 0, sizeof(selgame_Password));
 
 	selgame_FreeVectors();
 
@@ -385,6 +386,7 @@ void selgame_Password_Select(int value)
 {
 	if (selgame_selectedGame) {
 		setIniValue("Phone Book", "Entry1", selgame_Ip);
+		setIniValue("Phone Book", "Password1", selgame_Password);
 		if (SNetJoinGame(selgame_selectedGame, selgame_Ip, selgame_Password, NULL, NULL, gdwPlayerId)) {
 			if (!IsDifficultyAllowed(m_client_info->initdata->bDiff)) {
 				selgame_GameSelection_Select(1);
diff --git a/defs.h b/defs.h
index 059de08e..16e39d21 100644
--- a/defs.h
+++ b/defs.h
@@ -32,9 +32,9 @@
 #define MAXBELTITEMS			8
 #define MAXLIGHTS				32
 #define MAXMISSILES				125
-#define MAXMONSTERS				200
+#define MAXMONSTERS				400
 #define MAXMULTIQUESTS			4
-#define MAXOBJECTS				127
+#define MAXOBJECTS				254
 #define MAXPORTAL				4
 #define MAXQUESTS				16
 #define MAXTHEMES				50
@@ -59,7 +59,7 @@
 #define NUM_TOWNERS				16
 
 // todo: enums
-#define NUMLEVELS				17
+#define NUMLEVELS				34
 #define SMITH_ITEMS				20
 #define SMITH_PREMIUM_ITEMS		6
 #define SMITH_MAX_VALUE			140000
@@ -69,9 +69,9 @@
 #define MAXEXP					2000000000
 #define MAXRESIST				75
 
-#define GOLD_SMALL_LIMIT		1000
-#define GOLD_MEDIUM_LIMIT		2500
-#define GOLD_MAX_LIMIT			5000
+#define GOLD_SMALL_LIMIT		5000
+#define GOLD_MEDIUM_LIMIT		10000
+#define GOLD_MAX_LIMIT			20000
 
 #define PLR_NAME_LEN			32
 
